dozer-orchestrator/src/pipeline/conflict_resolver.rs

```diff
@@ -0,0 +1,105 @@
+use dozer_cache::cache::index::get_primary_key;
+use dozer_cache::errors::CacheError;
+use dozer_cache::errors::CacheError::PrimaryKeyExists;
+use dozer_types::types::{Operation, Record, Schema};
+
+use dozer_types::log::warn;
+use dozer_types::models::api_endpoint::{
+    ConflictResolution, OnDeleteResolutionTypes, OnInsertResolutionTypes, OnUpdateResolutionTypes,
+};
+
+pub struct ConflictResolver {}
+
+impl ConflictResolver {
+    pub fn resolve_insert_error(
+        new: Record,
+        schema: &Schema,
+        err: CacheError,
+        resolution: Option<ConflictResolution>,
```

Comment:
Maybe it's better to accept `OnInsertResolutionType` here and handle the defaults somewhere else?

Reply:
Yeah, refactored here and other 3 places